src/prefect/flow_engine.py

```diff
@@ -1369,7 +1370,9 @@ def run_flow_sync(
             with engine.run_context():
                 engine.call_flow_fn()
 
-    return engine.state if return_type == "state" else engine.result()
+    result = engine.result()
+    link_state_to_flow_run_result(engine.state, result)
```

Comment:
This is where the tracking is actually happening. The key is that the result is linked on the _parent flow run context_ (if it exists) and not this run's flow run context